demos/ota/ota_demo_core_mqtt/ota_demo_core_mqtt.c

```diff
@@ -0,0 +1,1354 @@
+                                    uint8_t qos );
+
+/**
+ * @brief Subscribe to the Mqtt topics.
+ *
+ * This function subscribes to the Mqtt topics with the Quality of service
+ * received as parameter. This function also registers a callback for the
+ * topicfilter.
+ *
+ * @param[in] pTopicFilter Mqtt topic filter.
+ *
+ * @param[in] topicFilterLength Length of the topic filter.
+ *
+ * @param[in] qos Quality of Service
+ *
+ * @param[in] callback Callback to be registered.
+ *
+ * @return OtaMqttSuccess if success , other error code on failure.
+ */
+static OtaMqttStatus_t mqttSubscribe( const char * pTopicFilter,
+                                      uint16_t topicFilterLength,
+                                      uint8_t qos,
+                                      OtaMqttCallback_t callback );
+
+/**
+ * @brief Unsubscribe to the Mqtt topics.
+ *
+ * This function unsubscribes to the Mqtt topics with the Quality of service
+ * received as parameter.
+ *
+ * @param[in] pTopicFilter Mqtt topic filter.
+ *
+ * @param[in] topicFilterLength Length of the topic filter.
+ *
+ * @param[qos] qos Quality of Service
+ *
+ * @return  OtaMqttSuccess if success , other error code on failure.
+ */
+static OtaMqttStatus_t mqttUnsubscribe( const char * pTopicFilter,
+                                        uint16_t topicFilterLength,
+                                        uint8_t qos );
+
+/**
+ * @brief Start OTA demo.
+ *
+ * @return   EXIT_SUCCESS or EXIT_FAILURE.
+ */
+static int startOTADemo( void );
+
+/**
+ * @brief Set OTA interfaces.
+ *
+ * @param[in]  pOtaInterfaces pointer to OTA interface structure.
+ *
+ * @return   None.
+ */
+static void setOtaInterfaces( OtaInterfaces_t * pOtaInterfaces );
+
+/**
+ * @brief Disconnect from the MQTT broker.
+ *
+ */
+static void disconnect( void );
+
+/**
+ * @brief Attempt to connect to the MQTT broker.
+ *
+ * @return int EXIT_SUCCESS if a connection is established.
+ */
+static int establishConnection( void );
+
+/**
+ * @brief Initialize MQTT by setting up transport interface and network.
+ *
+ * @param[in] pMqttContext Structure representing MQTT connection.
+ * @param[in] pNetworkContext Network context to connect on.
+ * @return int EXIT_SUCCESS if MQTT component is initialized
+ */
+static int initializeMqtt( MQTTContext_t * pMqttContext,
+                           NetworkContext_t * pNetworkContext );
+
+/**
+ * @brief Retry logic to establish a connection to the server.
+ *
+ * If the connection fails, keep retrying with exponentially increasing
+ * timeout value, until max retries, max timeout or successful connect.
+ *
+ * @param[in] pNetworkContext Network context to connect on.
+ * @return int EXIT_FAILURE if connection failed after retries.
+ */
+static int connectToServerWithBackoffRetries( NetworkContext_t * pNetworkContext );
+
+/**
+ * @brief Random number to be used as a back-off value for retrying connection.
+ *
+ * @return int32_t A random integer.
```

Comment:
```suggestion
 * @return uint32_t The generated random number.
```

Reply:
Addressed in https://github.com/aws/aws-iot-device-sdk-embedded-C/pull/1418/commits/726a7d066006ec340f26308d37075a414069f56b